### ISSUE TITLE
Layout/MultilineAssignmentLayout causes a false negative when using block on different line from left side of multi-line assignment

### ISSUE THREAD

**Issue by sanfrecce-osaka** [rubocop/rubocop#14880]
`Layout/MultilineAssignmentLayout` unrecognize block on different line from left side of multi-line assignment.

## Steps to reproduce the problem

```yaml
Layout/MultilineAssignmentLayout:
  Enabled: true
  EnforcedStyle: new_line
  SupportedTypes: ['block']
```

or

```yaml
Layout/MultilineAssignmentLayout:
  Enabled: true
  EnforcedStyle: same_line
  SupportedTypes: ['block']
```

## Expected behavior

The following code should be recognized:

### when new_line style

```ruby
bars = foo
^^^^^^^^^^ Right hand side of multi-line assignment is on the same line as the assignment operator `=`.
  .map { it.bar }
```

### when same_line style

```ruby
bars =
^^^^^^ Right hand side of multi-line assignment is not on the same line as the assignment operator `=`.
  foo.map { it.bar }
```

## Actual behavior

The above codes pass.

## RuboCop version

```
$ [bundle exec] rubocop -V
1.84.2 (using Parser 3.3.10.1, rubocop-ast 1.49.0, analyzing as Ruby 2.7, running on ruby 4.0.0) [arm64-darwin24]
  - rubocop-performance 1.26.1
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.9.0
```
